pandas/tests/indexes/common.py

```diff
@@ -682,6 +682,21 @@ def test_hasnans_isnans(self):
                 self.assert_numpy_array_equal(idx._isnan, expected)
                 self.assertTrue(idx.hasnans)
 
+    def test_ensure_copied_data(self):
+        # GH12309
+        for name, index in compat.iteritems(self.indices):
+            if isinstance(index, (PeriodIndex, RangeIndex, MultiIndex)):
+                # Cannot be instantiated from just _data as first argument
+                continue
+
+            data = index._data.copy()
+            idx_type = index.__class__
+            idx = idx_type(data, copy=True)
+            self.assert_numpy_array_equal(data, idx._data)
```

Comment:
Fine for ``tm.assert_index_equal``. What do you mean by "these are not copies"?! I check in ``test_copy()`` that objects are not the same. Here I am checking they don't have the same ``_data`` object (when ``copy=True``). I don't get how this relates to ``test_identical``, which [doesn't look at data](https://github.com/pydata/pandas/blob/c76ca444fd42675982e81394736d941e2f2629db/pandas/tests/indexes/test_numeric.py#L533).